### PR TITLE
Env var name change, update readme, test fix on one machine

### PR DESCRIPTION
**Pull request by xkazuka**
- [X] Allow make test run on a single machine
- [X] Deleted duplicate tests 
- [X] Change env var name where users can specify where the location of where credentials yaml and models will be downloaded (default .cache)
- [X] Update read me on the env var options
- [X] Bug fix where passed arguments were being overwritten by values in self

### CODE REVIEW

File: tests/sparsezoo/helpers.py [hunk 1]
```diff
@@ -23,11 +23,20 @@
 from sparsezoo.utils import CACHE_DIR
 
 
-def download_and_verify(model: str, other_args: Optional[Dict] = None):
+def download_and_verify(
+    model: str, test_name: Optional[str] = "", other_args: Optional[Dict] = None
```
Comment (rahul-tuli) [inline tests/sparsezoo/helpers.py:27]: Suggestion: Remove `Optional` here, `Optional` implies `None` can be passed as the `test_name`, and `str(None)` will be equal to `"None"` on line 33 
```suggestion
    model: str, test_name: str = "", other_args: Optional[Dict] = None
```

File: tests/sparsezoo/helpers.py [hunk 2]
```diff
@@ -23,11 +23,20 @@
...
+):
+    print(other_args)
     if other_args is None:
         other_args = {
-            "override_parent_path": os.path.join(CACHE_DIR, "test_download"),
+            "override_parent_path": os.path.join(
+                CACHE_DIR, "test_download", str(test_name)
```
Comment (rahul-tuli) [inline tests/sparsezoo/helpers.py:33]: Noting here `str` not needed if `Optional` is removed

File: tests/sparsezoo/helpers.py [hunk 3]
```diff
@@ -44,9 +53,12 @@ def model_constructor(
     sparse_name: str,
     sparse_category: str,
     sparse_target: Optional[str],
+    test_name: Optional[str],
```
Comment (rahul-tuli) [inline tests/sparsezoo/helpers.py:56]: Suggestion: Maybe default to empty string like above to maintain consistency

File: tests/sparsezoo/models/classification/test_efficientnet.py [hunk 4]
```diff
@@ -39,15 +42,16 @@ def test_efficientnet_b0(
...
+        constructor_function=efficientnet_b0,
+        download=download,
+        framework=framework,
+        repo=repo,
+        dataset=dataset,
+        training_scheme=training_scheme,
+        sparse_name=sparse_name,
+        sparse_category=sparse_category,
+        sparse_target=sparse_target,
+        test_name=os.path.join(str(os.path.basename(__file__)), str(datetime.now())),
```
Comment (rahul-tuli) [inline tests/sparsezoo/models/classification/test_efficientnet.py:54]: Nice!

Comment (rahul-tuli) [inline tests/sparsezoo/models/classification/test_efficientnet.py:54]: nit: `os.path.basename` returns `str` already
```suggestion
        test_name=os.path.join(os.path.basename(__file__), str(datetime.now())),
```

File: tests/sparsezoo/models/classification/test_efficientnet.py [hunk 5]
```diff
@@ -72,13 +76,14 @@ def test_efficientnet_b4(
...
+        constructor_function=efficientnet_b4,
+        download=download,
+        framework=framework,
+        repo=repo,
+        dataset=dataset,
+        training_scheme=training_scheme,
+        sparse_name=sparse_name,
+        sparse_category=sparse_category,
+        sparse_target=sparse_target,
+        test_name=os.path.join(str(os.path.basename(__file__)), str(datetime.now())),
```
Comment (rahul-tuli) [inline tests/sparsezoo/models/classification/test_efficientnet.py:88]: same as above!

File: tests/sparsezoo/models/test_zoo.py [hunk 6]
```diff
@@ -59,11 +60,18 @@
...
+            {
+                "override_parent_path": os.path.join(CACHE_DIR, "test_download"),
+            },
         ),
     ],
 )
 def test_load_model(model_args, other_args):
+    other_args["override_parent_path"] = os.path.join(
+        other_args["override_parent_path"],
+        str(os.path.basename(__file__)),
```
Comment (rahul-tuli) [inline tests/sparsezoo/models/test_zoo.py:72]:
```suggestion
        os.path.basename(__file__),
```

**Review by rahul-tuli — DISMISSED** [review]
Nice work, left a few cosmetic comments!

**Comment by rahul-tuli** [comment]
Nice Work! the code looks good, however I'm concerned about the repetition across many files, can we come up with a way to reduce that?

**Comment by kazukageorge** [comment]
> Nice Work! the code looks good, however I'm concerned about the repetition across many files, can we come up with a way to reduce that?

@rahul-tuli Where in the code? Need to gauge how much time it would take since the goal of the assignment is completed.

File: tests/sparsezoo/models/test_zoo.py [hunk 7]
```diff
@@ -683,25 +648,35 @@ def test_load_base_model_from_recipe(recipe_args, other_args):
...
 )
 def test_download_recipe_base_framework_files(recipe_args, other_args):
+    other_args["override_parent_path"] = os.path.join(
+        other_args["override_parent_path"],
+        os.path.basename(__file__),
+        str(datetime.now()),
+    )
     files = Zoo.download_recipe_base_framework_files(recipe_args, **other_args)
     assert len(files) > 0
+    # shutil.rmtree(str(other_args['o/verride_parent_path']))
```
Comment (rahul-tuli) [inline tests/sparsezoo/models/test_zoo.py:682]: Could we remove this?

**Comment by rahul-tuli** [comment]
> > Nice Work! the code looks good, however I'm concerned about the repetition across many files, can we come up with a way to reduce that?
> 
> @rahul-tuli Where in the code? Need to gauge how much time it would take since the goal of the assignment is completed.
```
test_resnet.py
Duplicate code: lines 205-305
test_vgg.py
Duplicate code: lines 33-99
Duplicate code: lines 206-306
test_zoo.py
Duplicate code: lines 49-66
Duplicate code: lines 103-120
Duplicate code: lines 388-396
Duplicate code: lines 417-425
Duplicate code: lines 451-469
Duplicate code: lines 505-523
Duplicate code: lines 537-574
Duplicate code: lines 586-623
```

However, whether it's a feasible exercise or/not based on priorities, I leave that upto you. Am good with the code once comments are addressed. Nice work!

File: tests/sparsezoo/helpers.py [hunk 8]
```diff
@@ -23,11 +23,18 @@
 from sparsezoo.utils import CACHE_DIR
 
 
-def download_and_verify(model: str, other_args: Optional[Dict] = None):
+def download_and_verify(
+    model: str, test_name: str = "", other_args: Optional[Dict] = None
+):
+    print(other_args)
```
Comment (dbarbuzzi) [inline tests/sparsezoo/helpers.py:29]: Is this `print` meant to be added/kept?

**Review by dbarbuzzi — DISMISSED** [review]

**Review by dbarbuzzi — APPROVED** [review]

**Review by tdg5 — APPROVED** [review]
Seems fine to me

**Pull request merged**